src/rgw/rgw_role.cc

```diff
@@ -0,0 +1,245 @@
+}
+
+int RGWRole::store_name(bool exclusive)
+{
+  RGWNameToId nameToId;
+  nameToId.obj_id = id;
+
+  string oid = get_names_oid_prefix() + name;
+
+  bufferlist bl;
+  ::encode(nameToId, bl);
+  return rgw_put_system_obj(store, store->get_zone_params().roles_pool, oid,
+              bl.c_str(), bl.length(), exclusive, NULL, real_time(), NULL);
+}
+
+int RGWRole::create(bool exclusive)
+{
+  int ret;
+
+  /* check to see the name is not used */
+  ret = read_id(name, id);
+  if (exclusive && ret == 0) {
+    ldout(cct, 0) << "ERROR: name " << name << " already in use for role id "
+                    << id << dendl;
+    return -EEXIST;
+  } else if ( ret < 0 && ret != -ENOENT) {
+    ldout(cct, 0) << "failed reading role id  " << id << ": "
+                  << cpp_strerror(-ret) << dendl;
+    return ret;
+  }
+
+  /* create unique id */
+  uuid_d new_uuid;
+  char uuid_str[37];
+  new_uuid.generate_random();
+  new_uuid.print(uuid_str);
+  id = uuid_str;
+
+  // Creation time
+  real_clock::time_point t = real_clock::now();
+
+  struct timeval tv;
+  real_clock::to_timeval(t, tv);
+
+  char buf[30];
+  strftime(buf,30,"%Y-%m-%dT%H:%M:%S", std::gmtime(&tv.tv_sec));
+  sprintf(buf + strlen(buf),".%dZ",(int)tv.tv_usec);
+  creation_date.assign(buf, strlen(buf));
+
+  auto& pool = store->get_zone_params().roles_pool;
+  ret = store_info(exclusive);
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR:  storing info for in pool: " << pool.name << ": "
+                  << id << ": " << cpp_strerror(-ret) << dendl;
+    return ret;
+  }
+
+  return store_name(exclusive);
+}
+
+int RGWRole::delete_obj()
+{
+  auto& pool = store->get_zone_params().roles_pool;
+
+  int ret = read_name();
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR: finding role name in pool: " << pool.name << ": "
+                  << name << ": " << cpp_strerror(-ret) << dendl;
+    return ret;
+  }
+
+  // Delete id
+  string oid = get_info_oid_prefix() + id;
+  ret = rgw_delete_system_obj(store, pool, oid, NULL);
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR: deleting role id from pool: " << pool.name << ": "
+                  << id << ": " << cpp_strerror(-ret) << dendl;
+  }
+
+  // Delete name
+  oid = get_names_oid_prefix() + name;
+  ret = rgw_delete_system_obj(store, pool, oid, NULL);
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR: deleting role name from pool: " << pool.name << ": "
+                  << name << ": " << cpp_strerror(-ret) << dendl;
+  }
+
+  return ret;
+}
+
+int RGWRole::get()
+{
+  auto& pool = store->get_zone_params().roles_pool;
+
+  int ret = read_name();
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR: finding role name in pool: " << pool.name << ": "
+                  << name << ": " << cpp_strerror(-ret) << dendl;
+    return ret;
+  }
+
+  ret = read_info();
+  if (ret < 0) {
+    ldout(cct, 0) << "ERROR: finding role id in pool: " << pool.name << ": "
+                  << id << ": " << cpp_strerror(-ret) << dendl;
```

Comment:
both `read_name()` and `read_info()` are already logging error messages, so i don't think we need to log it again here:
```
$ radosgw-admin role get --role-name=foo
2016-10-18 09:53:41.912480 7f959a04ba80  0 ERROR: failed reading role name from pool: default.rgw.roles: foo: (2) No such file or directory
could not get role: (2) No such file or directory
2016-10-18 09:53:41.912580 7f959a04ba80  0 ERROR: finding role name in pool: default.rgw.roles: foo: (2) No such file or directory
```